### PR TITLE
Merge UI into Dev

### PR DESCRIPTION
**Pull request by baumannkev**
Added ImGui windows functionality with dynamic resetting windows after every program shutdown.

### CODE REVIEW

**Comment by le-rich** [comment]
Future PRs should be signed with your name [KB] and your Linear issue ID [COM-XX]

File: .github/workflows/branch-and-pr-build-validate.yml
Comment (le-rich) [inline .github/workflows/branch-and-pr-build-validate.yml:1]: Was this included by accident?

File: Physics/test.txt
Comment (le-rich) [inline Physics/test.txt:1]: Lol

File: UI/main.cpp
Comment (le-rich) [inline UI/main.cpp:1]: This branch is significanty behind master, requesting that you pull latest and merge your changes with it. 

All other projects have also been moved into their own logic/ loops so please integrate into that latest architecture from @Sepehrman and @vinodbandla473

File: Physics/test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+piss on my dry balls
```
Comment (baumannkev) [inline Physics/test.txt:1]: that wasnt me 😆

**Review by le-rich — CHANGES_REQUESTED** [review]
Changes requested

**Review by le-rich — APPROVED** [review]

**Review by Tissure — DISMISSED** [review]
lgtm

**Review by Tissure — APPROVED** [review]
lgtm

File: Physics/test.txt (hunk 1 above)
Comment (imnorookie) [reply Physics/test.txt:1]: LOL sorry.

**Review by imnorookie — APPROVED** [review]
luWuks gUwUd two me!! :333

**Pull request merged**